### PR TITLE
feat: queue now uses RichDisplay

### PR DESCRIPTION
**Pull request by cfanoulis**
replaces #693

### CODE REVIEW

**Comment by cfanoulis** [comment]
informally asking @PyroTechniac to review as he was also reviewing on the old PR

File: src/commands/Music/queue.ts [hunk 1]
```diff
@@ -13,22 +15,46 @@ export default class extends MusicCommand {
...
+		const response = await message.send(new MessageEmbed()
+			.setColor(getColor(message))
+			.setDescription(message.language.tget('SYSTEM_LOADING')));
+
+		// Format the song entries
+		const songFields = await Promise.all(queue.map(async (song, pos) => message.language.tget('COMMAND_QUEUE_LINE', pos, Util.escapeMarkdown(song.title), song.url, song.friendlyDuration, Util.escapeMarkdown((await song.fetchRequester())!.username))));
+
+		// Generate the pages with 10 songs each
+		const pages = this.chunkify(songFields, 5);
+		const queueDisplay = new UserRichDisplay(new MessageEmbed()
```
Comment (favna) [inline src/commands/Music/queue.ts:32]: Do we really want Skyra's specific UserRichDisplay here? I'd rather say make it a regular klasa RichDisplay so anyone can navigate the queue.

File: src/commands/Music/queue.ts [hunk 2]
```diff
@@ -13,22 +15,46 @@ export default class extends MusicCommand {
...
+			.setColor(getColor(message))
+			.setTitle(message.language.tget('COMMAND_QUEUE_TITLE', message.guild!.name))
+			.addField(message.language.tget('COMMAND_QUEUE_NOWPLAYING_TITLE'), message.language.tget('COMMAND_QUEUE_NOWPLAYING', Util.escapeMarkdown(song.title), song.url, song.friendlyDuration, Util.escapeMarkdown((await song.fetchRequester())!.username)))
+			.addField(message.language.tget('COMMAND_QUEUE_TOTAL_TITLE'), message.language.tget('COMMAND_QUEUE_TOTAL', queue.length, showSeconds(queue.map(song => song.duration).reduce((a, b) => a + b)))));
+
+		for (const page of pages) {
+			queueDisplay.addPage((template: MessageEmbed) => template.setDescription(page.join('\n\n')));
```
Comment (favna) [inline src/commands/Music/queue.ts]:
```suggestion
			queueDisplay.addPage((embed: MessageEmbed) => embed.setDescription(page.join('\n\n')));
```

For consistency with other similar structures

File: src/commands/Music/queue.ts [hunk 3]
```diff
@@ -13,22 +15,46 @@ export default class extends MusicCommand {
...
+	 * @param array The array to convert into chunks
+	 * @param amount How many elements each chunk has
+	 */
+	private chunkify(array: string[], amount: number) {
+		if (array.length <= amount) return [array];
+		const chunks = [];
+		while (array.length > amount) {
+			chunks.push(array.splice(0, amount));
+		}
+		return chunks;
```
Comment (favna) [inline src/commands/Music/queue.ts]: `@klasa/utils` has a method to chunk an array. Made the same mistake a while back lmao, then Pyro pointed it out. Use that on please.

File: src/languages/es-ES.ts [hunk 4]
```diff
@@ -288,10 +288,15 @@ export default class extends Language {
...
-		COMMAND_QUEUE_TRUNCATED: amount => `Mostrando 10 canciones de ${amount}`,
+		COMMAND_QUEUE_DESCRIPTION: `Check the queue list.`,
+		COMMAND_QUEUE_EMPTY: `The session is over, add some songs to the queue, you can for example do \`Skyra, add Imperial March\`, and... *dumbrolls*!`,
+		COMMAND_QUEUE_LAST: `There are no more songs! After the one playing is over, the session will end!`,
+		COMMAND_QUEUE_TITLE: guildname => `Music queue for ${guildname}`,
+		COMMAND_QUEUE_LINE: (position, title, url, duration, requester) => `[${position}] [${title}](${url}) | (${duration}) | Requested by **${requester}**.`,
+		COMMAND_QUEUE_NOWPLAYING: (title, url, duration, requester) => `[${title}](${url}) | (${duration}) | Requested by **${requester}**.`,
+		COMMAND_QUEUE_NOWPLAYING_TITLE: 'Now Playing:',
+		COMMAND_QUEUE_TOTAL_TITLE: 'Total songs:',
+		COMMAND_QUEUE_TOTAL: (songs, remainingTime) => `${songs} song${songs === 1 ? '' : 's'} in the queue, with a total duration of ${remainingTime}`,
```
Comment (favna) [inline src/languages/es-ES.ts:299]: https://translate.google.com

**Review by favna — CHANGES_REQUESTED** [review]

File: src/commands/Music/queue.ts (hunk 1 above)
Comment (kyranet) [reply src/commands/Music/queue.ts:32]: Yes, we want to. To avoid trolls.

Comment (favna) [reply src/commands/Music/queue.ts:32]: fair enough, resolving.

File: src/languages/es-ES.ts (hunk 4 above)
Comment (cfanoulis) [reply src/languages/es-ES.ts:299]: Last time I used that, RAE and some kittens died. Skipping...

**Review by favna — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**